text/0004-structured_data_io.md

```diff
@@ -0,0 +1,215 @@
+- Feature Name: structured_io
+- Start Date: 2017-10-18
+- RFC PR: [#4](https://github.com/qri-io/rfcs/pull/4)
+- Issue: N/A
+- Repo: [dataset](https://github.com/qri-io/dataset)
+
+# Summary
+[summary]: #summary
+
+Structured I/O defines interfaces for reading & writing streams of _entries_,
+which are elements of a composite data structure such as an array or object.
+Structured reader & writer interfaces combine a byte stream, data format 
+and schema to create entry readers & writers that produce & consume entries
+instead of bytes. Structured I/O streams can be composed & connected to form
+the basis of rich data communication capable of spanning across formats.
+
+# Motivation
+[motivation]: #motivation
+
+Structured I/O is the primary means of _doing things with data_. All of the
```

Comment:
Obviously Structured I/O is about input and output but I misread this sentence and expected that it was also intended to provide higher level facilities for "doing things with data".  not that it needs to be changed but just flagging that

Reply:
Great point. "Doing things with data" is getting ahead of ourselves, and positions this API as being high-level, when it's in fact a lower-level parsing API. That has me thinking about a higher-level API? We've talked about a `stats` package & statistical metadata for a while, maybe it's time to think about an RFC for a package based around structured I/O primitives that holds common statistical functions?